### PR TITLE
Added GPU backend for DBSCAN (#481)

### PR DESCRIPTION
**Pull request by amgrigoriev**
-
-

### CODE REVIEW

File: examples/cpp_sycl/makefile_lnx [hunk 1]
```diff
@@ -116,7 +116,7 @@ sointel64:
 _make_ex: $(RES)
 
 vpath
-vpath %.cpp $(addprefix ./source/,datasource covariance linear_regression logistic_regression optimization_solvers pca pca_transform kmeans moments k_nearest_neighbors gradient_boosted_trees)
+vpath %.cpp $(addprefix ./source/,datasource covariance linear_regression logistic_regression optimization_solvers pca pca_transform kmeans moments k_nearest_neighbors gradient_boosted_trees dbscan)
```
Comment (michael-smirnov) [inline examples/cpp_sycl/makefile_lnx:119]: Don't we need to have this list sorted alphabetically?

Comment (amgrigoriev) [reply examples/cpp_sycl/makefile_lnx:119]: Almost every method is added at the end for now. All are in historical order, I guess

Comment (michael-smirnov) [reply examples/cpp_sycl/makefile_lnx:119]: No, at the beginning all methods were sorted: problems started from `k_nearest_neighbors`. It is not critical but should be fixed later for both master and releases

**Pull request merged**